src/rpc/methods/eth.rs

```diff
@@ -2015,19 +2015,51 @@ pub enum EthGetTransactionByBlockNumberAndIndex {}
 impl RpcMethod<2> for EthGetTransactionByBlockNumberAndIndex {
     const NAME: &'static str = "Filecoin.EthGetTransactionByBlockNumberAndIndex";
     const NAME_ALIAS: Option<&'static str> = Some("eth_getTransactionByBlockNumberAndIndex");
-    const PARAM_NAMES: [&'static str; 2] = ["p1", "p2"];
+    const PARAM_NAMES: [&'static str; 2] = ["block_number", "tx_index"];
     const API_PATHS: ApiPaths = ApiPaths::V1;
     const PERMISSION: Permission = Permission::Read;
 
     type Params = (EthUint64, EthUint64);
     type Ok = Option<ApiEthTx>;
 
     async fn handle(
-        _ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
-        (_p1, _p2): Self::Params,
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (block_number, tx_index): Self::Params,
     ) -> Result<Self::Ok, ServerError> {
-        // Lotus doesn't support this method (v1.29.0), so do we.
-        Err(ServerError::unsupported_method())
+        let height = block_number.0 as ChainEpoch;
+        let head = ctx.chain_store().heaviest_tipset();
+
+        if height > head.epoch() {
```

Comment:
What about using `ensure` macro?

https://docs.rs/anyhow/latest/anyhow/macro.ensure.html

Reply:
Sorry, forget about this one. I don’t think using a `bail!` or `ensure!` macro is possible here.